### PR TITLE
docs: Added DSPy integration docs

### PR DESCRIPTION
**Pull request by Anush008**
This PR intends to add documentation about the https://github.com/stanfordnlp/dspy integration.

Section preview: https://deploy-preview-390--condescending-goldwasser-91acf0.netlify.app/documentation/integrations/dspy/

### CODE REVIEW

**Comment by Anush008** [comment]
@kacperlukawski, review 🙏.

File: qdrant-landing/content/documentation/integrations/dspy.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+We can configure `DSPy` settings to use the Qdrant retriever model like so:
+```python
+import dspy
+from dspy.retrieve.qdrant_rm import QdrantRM
+
+from qdrant_client import QdrantClient
+
+turbo = dspy.OpenAI(model='gpt-3.5-turbo')
+qdrant_client = Qdrant()    # Defaults to a local instance at http://localhost:6333/
```
Comment (kacperlukawski) [inline qdrant-landing/content/documentation/integrations/dspy.md:33]: Shouldn't it be `QdrantClient`?

**Review by kacperlukawski — CHANGES_REQUESTED** [review]
There is an issue in the code snippet that has to be fixed. Moreover, the code is inconsistent - single and double quotes are mixed.

File: qdrant-landing/content/documentation/integrations/dspy.md (hunk 1 above)
Comment (Anush008) [reply qdrant-landing/content/documentation/integrations/dspy.md]: https://github.com/qdrant/landing_page/pull/390/commits/5475fa800ee3badab782b970f39026729cc686ab.
Fixed.

**Review by kacperlukawski — APPROVED** [review]

**Pull request merged**